<launch>
    <arg name="bag_path" default="/home/uib/bagfiles/cabrera/2025_10_28/14_44_35/"/> 
    <!-- <arg name="bag_name" default="sparus2_multibeam_2025-11-10-10-18-53_0.bag"/> -->
    <arg name="bag_name" default="sparus2_multibeam_2025-10-28-14-44-35_0.bag"/>
    
    <!-- <arg name="bag_name" default="sparus2_multibeam_2025-11-10-11-31-34_01.bag"/> -->
    <arg name="output_dir" default="/home/uib/derelictes_ws/bagfiles/results/"/>
    
    <arg name="offset_x" default="0.0"/>
    <arg name="offset_y" default="0.0"/>
    <arg name="offset_z" default="0.0"/>
    <arg name="yaw_deg"  default="90.0"/>

    <arg name="sor_k" default="50"/>
    <arg name="sor_std" default="1.0"/>
    <arg name="voxel_size" default="0.1"/>

    <arg name="enable_slam" default="true"/>
    <arg name="slam_map_voxel" default="0.5"/>
    <arg name="slam_gicp_dist" default="2.0"/>

    <param name="use_sim_time" value="true"/>
    
    <node pkg="pointcloud_lib" type="multibeam_processor.py" name="mb_multi_proc" output="screen" required="true">
        <param name="bag_file"      value="$(arg bag_path)$(arg bag_name)" />
        <param name="scan_topic"    value="/sparus2/norbit_wbms_multibeam/multibeam_scan" />
        <param name="nav_topic"     value="/sparus2/navigator/navigation" />
        <param name="output_dir"    value="$(arg output_dir)" />
        
        <param name="offset_x" value="$(arg offset_x)" />
        <param name="offset_y" value="$(arg offset_y)" />
        <param name="offset_z" value="$(arg offset_z)" />
        <param name="sensor_yaw" value="$(eval arg('yaw_deg') * 3.14159265 / 180.0)" />
        
        <param name="sor_k" value="$(arg sor_k)"/>
        <param name="sor_std" value="$(arg sor_std)"/>
        <param name="voxel_size" value="$(arg voxel_size)"/>
        
        <param name="enable_slam" value="$(arg enable_slam)"/>
        <param name="slam_map_voxel" value="$(arg slam_map_voxel)"/>
        <param name="slam_gicp_dist" value="$(arg slam_gicp_dist)"/>
    </node>
</launch>